src/app/components/model3D/glb-viewer/glb-viewer.component.ts

```diff
@@ -0,0 +1,346 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  input,
+  OnDestroy,
+  OnInit,
+  viewChild,
+  inject,
+  effect,
+} from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { MatIcon } from '@angular/material/icon';
+import * as THREE from 'three';
+import { GLTFLoader } from 'three/examples/jsm/loaders/GLTFLoader.js';
+import { OrbitControls } from 'three/examples/jsm/controls/OrbitControls.js';
+import { Constants } from '@common/constants/constants';
+import { I18nService } from '@services/general';
+import { TranslatePipe } from '@ngx-translate/core';
+import { Model3DCreateHttpService, Model3DAnimalHttpService } from '@services/http';
+/**component for 3D model visalization
+ * @autor nav
+ * */
+
+@Component({
+  selector: 'app-glb-viewer',
+  imports: [CommonModule, MatIcon, TranslatePipe],
+  templateUrl: './glb-viewer.component.html',
+  styleUrl: './glb-viewer.component.scss',
+  changeDetection: Constants.changeDetectionStrategy,
+})
+export class GlbViewerComponent implements OnInit, OnDestroy {
+  readonly animalId = input<number>();
+  readonly width = input<string>('100%');
+  readonly height = input<string>('100%');
+  readonly autoRotate = input<boolean>(true);
+  isLoading = true;
+
+  errorMessage: string | null = null;
+  private readonly canvas = viewChild<ElementRef<HTMLCanvasElement>>('canvas');
+  private scene!: THREE.Scene;
+  private camera!: THREE.PerspectiveCamera;
+  private renderer!: THREE.WebGLRenderer;
+  private controls!: OrbitControls;
+  private animationId: number | null = null;
+  private loader = new GLTFLoader();
+  private currentModelUrl: string | null = null;
+  readonly model3DAnimalHttpService = inject(Model3DAnimalHttpService);
+  readonly i18nService = inject(I18nService);
+
+  constructor() {
```

Comment:
Es buena practica hacer que los efectos se lleven a una propiedad de clase. (Funcionan igual si estan dentro del constructor, y como es una propiedad de clase podria ponerle yo un nombre a la variable indicando que hace y su respectivo jsdoc.


```suggestion
  /**
  *Aqui mi jsdoc que le indico al efecto que es lo que hace para que otro
  * desarrollador entienda que es lo que hace el efecto.
  * @author dgutierrez
  */
  private readonly miEfectoQueHaceAlgo = effect(() => {
     aqui la logica del efecto
  })
```